src/statistics.jl

```diff
@@ -500,67 +500,220 @@ function apply_statistic(stat::Density2DStatistic,
+    performance.
+    """
     n::Int
-    # Bandwidth used for the kernel density estimation
+
+    """
+    Smoothing bandwidth used for the kernel density estimation. This
+    corresponds to the standard deviation of the `kernel`.
+    """
     bw::Real
+
+    """
+    Multiplicative adjustment of the computed optimal bandwidth. This is a
+    relative adjustment, see `bw` to enforce a specific numerical bandwidth.
+    """
+    adjust::Float64
+
+    """
+    Kernel used for density estimation, see `KernelDensity.jl` for more details.
+    Default is the Normal Distribution.
+    """
+    kernel
+
+    """
+    This parameter only applies in the context of multiple densities. If set to
+    `false` (the default), the densities are computed over the full range of
+    data. If `true`, then each density's range will be computed only over the
+    range of data belonging to that group. This option is incompatible with
+    stacked densities since the ranges might not line up any more.
+    """
+    trim::Bool
+
+    """
+    Method for scaling across multiple estimates. If `:area` (default), all
+    density estimates will have the same area under the curve (prior to trimming
+    ). If `:count`, the areas are scaled proportionally to the total number of
+    observations for each density estimate. If `:peak`, then all densities will
+    have the same maximum peak height.
+    """
+    scale::Symbol
+
+    """
+    Control handling of multiple overlapping densities. The `:dodge` option
+    (default) just overlays each density such that they are in front of each
+    other. The `:stack` option places the densities a top of each other. The
+    `:fill` option is similar to `:stack`, but the stacks are all normalized to
+    a constant height of 1.0. This last option is useful for generating
+    conditional density estimates.
+    """
+    position::Symbol
+
+    """
+    Whether the plot is `:horizontal` or `:vertical`
+    """
+    orientation::Symbol
+
+    """
+    Internal flag that is `true` if this density statistic is a violin plot
+    """
+    isviolin::Bool
 end
-DensityStatistic(; n=256, bandwidth=-Inf) = DensityStatistic(n, bandwidth)
 
-input_aesthetics(stat::DensityStatistic) = [:x, :color]
-output_aesthetics(stat::DensityStatistic) = [:x, :y, :color]
-default_scales(::DensityStatistic) = [Gadfly.Scale.y_continuous()]
+function input_aesthetics(stat::DensityStatistic)
+    if stat.isviolin
+        return [:x, :y, :color]
+    elseif stat.orientation == :horizontal
+        return [:x, :color]
+    else
+        return [:y, :color]
+    end
+end
 
-"""
-    Stat.density[(; n=256, bandwidth=-Inf)]
+output_aesthetics(stat::DensityStatistic) = [:x, :y, :color]
 
-Estimate the density of `x` at `n` points, and put the result in `x` and `y`.
-Smoothing is controlled by `bandwidth`.  Used by [`Geom.density`](@ref Gadfly.Geom.density).
-"""
-const density = DensityStatistic
```

Comment:
to be consistent with the rest of the API, i'd suggest keeping `const density = DensityStatistic` and moving the docstrings for `struct DensityStatistic` to the retained`Stat.density`.

Reply:
Will do